dnscrypt-proxy/dnsutils.go

```diff
@@ -459,7 +459,11 @@ func _dnsExchange(
 		if err != nil {
 			return DNSExchangeResponse{err: err}
 		}
-		defer pc.Close()
+		defer func(pc net.Conn) {
+			if pc != nil {
```

Comment:
The only possible way I can see this being `nil` here is if `net.DialTimeout` returned `nil, nil`. If we don't trust `net.DialTimeout` then it's simpler to check at the same time as `if err != nil` above.